namespaces/live.cloud-platform.service.justice.gov.uk/data-platform-datahub-catalogue-dev/05-prometheusrule.yaml

```diff
@@ -0,0 +1,299 @@
+          for: 15m
+          labels:
+            severity: DataHub-dev
+        - alert: ContainerRestartAlert
+          annotations:
+            message: :alert:Service '{{"{{" }} $labels.pod {{"}}"}}' has restarted.
+            runbook_url: https://runbooks.data-catalogue.service.justice.gov.uk/
+            dashboard_url: https://grafana.live.cloud-platform.service.justice.gov.uk/d/2yTpJtUU4G8iGuqPdEkG/datahub-deployment-status-dashboard
+          expr: (sum(increase(kube_pod_container_status_restarts_total{namespace="data-platform-datahub-catalogue-dev"}[10m])) by (container, namespace)) > 0
+          for: 5m
+          labels:
+            severity: DataHub-dev
+        - alert: KubePodCrashLooping
+          annotations:
+            message: >-
+              Pod {{ $labels.namespace }}/{{ $labels.pod }} ({{ $labels.container
+              }}) is restarting {{ printf "%.2f" $value }} times / 5 minutes.
+            runbook_url: https://runbooks.data-catalogue.service.justice.gov.uk/
+            dashboard_url: https://grafana.live.cloud-platform.service.justice.gov.uk/d/2yTpJtUU4G8iGuqPdEkG/datahub-deployment-status-dashboard
+          expr: >-
+            rate(kube_pod_container_status_restarts_total{namespace="data-platform-datahub-catalogue-dev", job="kube-state-metrics"}[15m])
+            * 60 * 5 > 0
+          for: 1h
+          labels:
+            severity: DataHub-dev
+        - alert: KubePodNotReady
+          annotations:
+            message: >-
+              Pod {{ $labels.namespace }}/{{ $labels.pod }} has been in a non-ready
+              state for longer than an hour.
+            runbook_url: https://runbooks.data-catalogue.service.justice.gov.uk/
+            dashboard_url: https://grafana.live.cloud-platform.service.justice.gov.uk/d/2yTpJtUU4G8iGuqPdEkG/datahub-deployment-status-dashboard
+          expr: >-
+            sum by (namespace, pod) (kube_pod_status_phase{namespace="data-platform-datahub-catalogue-dev", job="kube-state-metrics",
+            phase=~"Pending|Unknown"}) > 0
+          for: 1h
+          labels:
+            severity: DataHub-dev
+
+    - name: resource-alert
+      rules:
+        - alert: GMSCpuUsageHigh
+          annotations:
+            message: :warning:Service '{{"{{" }} $labels.container {{"}}"}}' cpu usage above threshold of 60%.
+            runbook_url: https://runbooks.data-catalogue.service.justice.gov.uk/
+            dashboard_url: https://grafana.live.cloud-platform.service.justice.gov.uk/d/2yTpJtUU4G8iGuqPdEkG/datahub-deployment-status-dashboard
+          expr: >-
+            (sum(rate(container_cpu_usage_seconds_total{namespace="data-platform-datahub-catalogue-dev", container=~".+gms"}[5m])) 
+            / sum(kube_pod_container_resource_limits{namespace="data-platform-datahub-catalogue-dev", container=~".+gms", unit="core"})) * 100  
+            > 90
```

Comment:
looking through this to grab same metrics for find-moj-data, should this be 60 or message be 90?

Reply:
good spot - the two need to match, but it's a judgement call based on unusual usage patterns (I've gone for 90% for the DataHub gms container)